packages/react-components/src/components/abstract-output-component.tsx

```diff
@@ -108,8 +109,22 @@ export abstract class AbstractOutputComponent<P extends AbstractOutputProps, S e
         return this.props.style.handleWidth;
     }
 
+    private checkAndRenderMainArea(): React.ReactNode {
+        if (this.state.outputStatus === ResponseStatus.FAILED) {
+            return this.analysisFailedMessage();
+        }
+
+        if (this.state.outputStatus === ResponseStatus.COMPLETED && this.resultsAreEmpty()) {
+            return this.emptyResultsMessage();
+        }
+
+        return this.renderMainArea();
+    }
+
     abstract renderMainArea(): React.ReactNode;
 
+    abstract resultsAreEmpty(): boolean;
+
     protected analysisFailedMessage(): React.ReactFragment {
```

Comment:
We suggest to rename this method and the next to renderAnalysisFailed() and renderEmptyResults(). Just in case a subclass overrides to show something else than a message.

Reply:
My reflex tends to be to make things more general when it's needed rather than a priori, but I don't have very strong feelings about it in this case. It will mainly take a little more code reading to understand what these functions do.

I will make these changes!